selfdrive/hardware/tici/agnos.py

```diff
@@ -33,6 +40,13 @@ def read(self, length: int) -> bytes:
         break
       out = self.decompressor.decompress(compressed)
       self.buf += out
+      self.downloaded += len(out)
+      now = time.monotonic()
+      if (now - self.last_read > 1):
+        percent_done = round(self.downloaded / self.file_size * 100)
```

Comment:
This is now wrong since the data is compressed (and sparsified). The request content length is not equal to the size of the output from the decompressor and unsparsifier.

To measure download speed you need to check the amount of input bytes processed before the decompressor.